From 2b9f50b69d75bc3e495b5b164eadb71864119ed1 Mon Sep 17 00:00:00 2001
From: Lorenzo Gaifas <brisvag@gmail.com>
Date: Wed, 4 Oct 2023 21:54:14 +0200
Subject: [PATCH] Vispy 0.14 (#6214)

Just a simple bump to our vispy dependency, now that 0.14 is out. This
comes with a few fixes and new features relevant for napari:

- Fix for wrong spherical marker depth:
https://github.com/vispy/vispy/pull/2506
- Fix for wrong shading of surfaces with flipped normals:
https://github.com/vispy/vispy/pull/2493
- picking filters for mesh and markers by @aganders, which can now be
used in napari to drastically speed up our get-status:
https://github.com/vispy/vispy/pull/2500

---------

Co-authored-by: Grzegorz Bokota <bokota+github@gmail.com>
Co-authored-by: napari-bot <napari-bot@users.noreply.github.com>
---
 resources/constraints/constraints_py3.10.txt  | 293 ++++++++---------
 resources/constraints/constraints_py3.11.txt  | 287 ++++++++---------
 resources/constraints/constraints_py3.8.txt   | 264 ++++++++--------
 resources/constraints/constraints_py3.9.txt   | 294 ++++++++---------
 .../constraints_py3.9_examples.txt            | 298 +++++++++---------
 setup.cfg                                     |   2 +-
 6 files changed, 729 insertions(+), 709 deletions(-)

diff --git a/resources/constraints/constraints_py3.10.txt b/resources/constraints/constraints_py3.10.txt
index b034be78..6c36a17e 100644
--- a/resources/constraints/constraints_py3.10.txt
+++ b/resources/constraints/constraints_py3.10.txt
@@ -2,60 +2,60 @@
 # This file is autogenerated by pip-compile with Python 3.10
 # by the following command:
 #
-#    pip-compile --allow-unsafe --extra=pyqt5 --extra=pyqt6_experimental --extra=pyside2 --extra=pyside6_experimental --extra=testing --output-file=resources/constraints/constraints_py3.10.txt --strip-extras resources/constraints/version_denylist.txt setup.cfg
+#    pip-compile --allow-unsafe --extra=pyqt5 --extra=pyqt6_experimental --extra=pyside2 --extra=pyside6_experimental --extra=testing --output-file=napari_repo/resources/constraints/constraints_py3.10.txt --strip-extras napari_repo/resources/constraints/version_denylist.txt napari_repo/setup.cfg
 #
 alabaster==0.7.13
     # via sphinx
-app-model==0.2.0
-    # via napari (setup.cfg)
+app-model==0.2.2
+    # via napari (napari_repo/setup.cfg)
 appdirs==1.4.4
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   npe2
 asciitree==0.3.3
     # via zarr
-asttokens==2.2.1
+asttokens==2.4.0
     # via stack-data
 attrs==23.1.0
     # via
     #   hypothesis
     #   jsonschema
     #   referencing
-babel==2.12.1
+babel==2.13.0
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   sphinx
 backcall==0.2.0
     # via ipython
-build==0.10.0
+build==1.0.3
     # via npe2
 cachey==0.2.1
-    # via napari (setup.cfg)
+    # via napari (napari_repo/setup.cfg)
 certifi==2023.7.22
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   requests
-charset-normalizer==3.2.0
+charset-normalizer==3.3.0
     # via requests
-click==8.1.6
+click==8.1.7
     # via
     #   dask
     #   typer
 cloudpickle==2.2.1
     # via dask
-cmake==3.27.1
+cmake==3.27.6
     # via triton
 comm==0.1.4
     # via ipykernel
-contourpy==1.1.0
+contourpy==1.1.1
     # via matplotlib
-coverage==7.2.7
+coverage==7.3.2
     # via pytest-cov
-cycler==0.11.0
+cycler==0.12.0
     # via matplotlib
-dask==2023.8.0
-    # via napari (setup.cfg)
-debugpy==1.6.7
+dask==2023.9.3
+    # via napari (napari_repo/setup.cfg)
+debugpy==1.8.0
     # via ipykernel
 decorator==5.1.1
     # via ipython
@@ -63,42 +63,43 @@ distlib==0.3.7
     # via virtualenv
 docstring-parser==0.15
     # via magicgui
-docutils==0.17.1
+docutils==0.20.1
     # via sphinx
 entrypoints==0.4
     # via numcodecs
-exceptiongroup==1.1.2
+exceptiongroup==1.1.3
     # via
     #   hypothesis
+    #   ipython
     #   pytest
-executing==1.2.0
+executing==2.0.0
     # via stack-data
-fasteners==0.18
+fasteners==0.19
     # via zarr
-filelock==3.12.2
+filelock==3.12.4
     # via
     #   torch
     #   triton
     #   virtualenv
-fonttools==4.42.0
+fonttools==4.43.0
     # via matplotlib
 freetype-py==2.4.0
     # via vispy
-fsspec==2023.6.0
+fsspec==2023.9.2
     # via
     #   dask
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
 heapdict==1.0.1
     # via cachey
-hsluv==5.0.3
+hsluv==5.0.4
     # via vispy
-hypothesis==6.82.2
-    # via napari (setup.cfg)
+hypothesis==6.87.1
+    # via napari (napari_repo/setup.cfg)
 idna==3.4
     # via requests
-imageio==2.31.1
+imageio==2.31.5
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   napari-svg
     #   scikit-image
 imagesize==1.4.1
@@ -109,59 +110,59 @@ in-n-out==0.1.8
     # via app-model
 iniconfig==2.0.0
     # via pytest
-ipykernel==6.25.0
+ipykernel==6.25.2
     # via
     #   napari-console
     #   qtconsole
-ipython==8.14.0
+ipython==8.16.1
     # via
     #   ipykernel
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   napari-console
 ipython-genutils==0.2.0
     # via qtconsole
-jedi==0.19.0
+jedi==0.19.1
     # via ipython
 jinja2==3.1.2
     # via
     #   numpydoc
     #   sphinx
     #   torch
-jsonschema==4.19.0
-    # via napari (setup.cfg)
+jsonschema==4.19.1
+    # via napari (napari_repo/setup.cfg)
 jsonschema-specifications==2023.7.1
     # via jsonschema
-jupyter-client==8.3.0
+jupyter-client==8.3.1
     # via
     #   ipykernel
     #   qtconsole
-jupyter-core==5.3.1
+jupyter-core==5.3.2
     # via
     #   ipykernel
     #   jupyter-client
     #   qtconsole
-kiwisolver==1.4.4
+kiwisolver==1.4.5
     # via
     #   matplotlib
     #   vispy
 lazy-loader==0.3
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   scikit-image
-lit==16.0.6
+lit==17.0.2
     # via triton
 locket==1.0.0
     # via partd
 lxml==4.9.3
-    # via napari (setup.cfg)
-magicgui==0.7.2
-    # via napari (setup.cfg)
+    # via napari (napari_repo/setup.cfg)
+magicgui==0.7.3
+    # via napari (napari_repo/setup.cfg)
 markdown-it-py==3.0.0
     # via rich
 markupsafe==2.1.3
     # via jinja2
-matplotlib==3.7.2
-    # via napari (setup.cfg)
+matplotlib==3.8.0
+    # via napari (napari_repo/setup.cfg)
 matplotlib-inline==0.1.6
     # via
     #   ipykernel
@@ -173,17 +174,17 @@ mpmath==1.3.0
 mypy-extensions==1.0.0
     # via psygnal
 napari-console==0.0.8
-    # via napari (setup.cfg)
+    # via napari (napari_repo/setup.cfg)
 napari-plugin-engine==0.2.0
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   napari-console
     #   napari-svg
 napari-plugin-manager==0.1.0a2
-    # via napari (setup.cfg)
+    # via napari (napari_repo/setup.cfg)
 napari-svg==0.1.10
-    # via napari (setup.cfg)
-nest-asyncio==1.5.7
+    # via napari (napari_repo/setup.cfg)
+nest-asyncio==1.5.8
     # via ipykernel
 networkx==3.1
     # via
@@ -191,21 +192,20 @@ networkx==3.1
     #   torch
 npe2==0.7.2
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   napari-plugin-manager
 numcodecs==0.11.0
     # via zarr
-numpy==1.25.2
+numpy==1.26.0
     # via
     #   contourpy
     #   dask
     #   imageio
     #   matplotlib
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   napari-svg
     #   numcodecs
     #   pandas
-    #   pywavelets
     #   scikit-image
     #   scipy
     #   tensorstore
@@ -213,8 +213,8 @@ numpy==1.25.2
     #   vispy
     #   xarray
     #   zarr
-numpydoc==1.5.0
-    # via napari (setup.cfg)
+numpydoc==1.6.0
+    # via napari (napari_repo/setup.cfg)
 nvidia-cublas-cu11==11.10.3.66
     # via
     #   nvidia-cudnn-cu11
@@ -240,7 +240,7 @@ nvidia-nccl-cu11==2.14.3
     # via torch
 nvidia-nvtx-cu11==11.7.91
     # via torch
-packaging==23.1
+packaging==23.2
     # via
     #   build
     #   dask
@@ -255,103 +255,101 @@ packaging==23.1
     #   superqt
     #   vispy
     #   xarray
-pandas==2.0.3 ; python_version >= "3.9"
+pandas==2.1.1 ; python_version >= "3.9"
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   xarray
 parso==0.8.3
     # via jedi
-partd==1.4.0
+partd==1.4.1
     # via dask
 pexpect==4.8.0
     # via ipython
 pickleshare==0.7.5
     # via ipython
-pillow==10.0.0
+pillow==10.0.1
     # via
     #   imageio
     #   matplotlib
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   scikit-image
 pint==0.22
-    # via napari (setup.cfg)
-platformdirs==3.10.0
+    # via napari (napari_repo/setup.cfg)
+platformdirs==3.11.0
     # via
     #   jupyter-core
     #   pooch
     #   virtualenv
-pluggy==1.2.0
+pluggy==1.3.0
     # via pytest
 pooch==1.7.0
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   scikit-image
 prompt-toolkit==3.0.39
     # via ipython
 psutil==5.9.5
     # via
     #   ipykernel
-    #   napari (setup.cfg)
-psygnal==0.9.1
+    #   napari (napari_repo/setup.cfg)
+psygnal==0.9.4
     # via
     #   app-model
     #   magicgui
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   npe2
 ptyprocess==0.7.0
     # via pexpect
 pure-eval==0.2.2
     # via stack-data
-pydantic==1.10.12
+pydantic==1.10.13
     # via
     #   app-model
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   npe2
 pygments==2.16.1
     # via
     #   ipython
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   qtconsole
     #   rich
     #   sphinx
     #   superqt
 pyopengl==3.1.6
     # via
-    #   -r resources/constraints/version_denylist.txt
-    #   napari (setup.cfg)
-pyparsing==3.0.9
+    #   -r napari_repo/resources/constraints/version_denylist.txt
+    #   napari (napari_repo/setup.cfg)
+pyparsing==3.1.1
     # via matplotlib
 pyproject-hooks==1.0.0
     # via build
 pyqt5==5.15.9
-    # via napari (setup.cfg)
+    # via napari (napari_repo/setup.cfg)
 pyqt5-qt5==5.15.2
     # via pyqt5
 pyqt5-sip==12.12.2
     # via pyqt5
 pyqt6==6.5.2
-    # via napari (setup.cfg)
+    # via napari (napari_repo/setup.cfg)
 pyqt6-qt6==6.5.2
     # via pyqt6
 pyqt6-sip==13.5.2
     # via pyqt6
-pyrsistent==0.19.3
-    # via jsonschema
 pyside2==5.15.2.1 ; python_version != "3.8"
-    # via napari (setup.cfg)
+    # via napari (napari_repo/setup.cfg)
 pyside6==6.4.2 ; python_version >= "3.10"
     # via
-    #   -r resources/constraints/version_denylist.txt
-    #   napari (setup.cfg)
+    #   -r napari_repo/resources/constraints/version_denylist.txt
+    #   napari (napari_repo/setup.cfg)
 pyside6-addons==6.4.2
     # via pyside6
 pyside6-essentials==6.4.2
     # via
     #   pyside6
     #   pyside6-addons
-pytest==7.4.0
+pytest==7.4.2
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   pytest-cov
     #   pytest-json-report
     #   pytest-metadata
@@ -359,43 +357,41 @@ pytest==7.4.0
     #   pytest-qt
 pytest-cov==4.1.0
     # via
-    #   -r resources/constraints/version_denylist.txt
-    #   napari (setup.cfg)
+    #   -r napari_repo/resources/constraints/version_denylist.txt
+    #   napari (napari_repo/setup.cfg)
 pytest-json-report==1.5.0
-    # via -r resources/constraints/version_denylist.txt
+    # via -r napari_repo/resources/constraints/version_denylist.txt
 pytest-metadata==3.0.0
     # via pytest-json-report
 pytest-pretty==1.2.0
-    # via napari (setup.cfg)
+    # via napari (napari_repo/setup.cfg)
 pytest-qt==4.2.0
-    # via napari (setup.cfg)
+    # via napari (napari_repo/setup.cfg)
 python-dateutil==2.8.2
     # via
     #   jupyter-client
     #   matplotlib
     #   pandas
-pytz==2023.3
+pytz==2023.3.post1
     # via pandas
-pywavelets==1.4.1
-    # via scikit-image
 pyyaml==6.0.1
     # via
     #   dask
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   npe2
-pyzmq==25.1.0
+pyzmq==25.1.1
     # via
     #   ipykernel
     #   jupyter-client
     #   qtconsole
-qtconsole==5.4.3
+qtconsole==5.4.4
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   napari-console
-qtpy==2.3.1
+qtpy==2.4.0
     # via
     #   magicgui
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   napari-console
     #   napari-plugin-manager
     #   qtconsole
@@ -408,20 +404,20 @@ requests==2.31.0
     # via
     #   pooch
     #   sphinx
-rich==13.5.2
+rich==13.6.0
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   npe2
     #   pytest-pretty
-rpds-py==0.9.2
+rpds-py==0.10.3
     # via
     #   jsonschema
     #   referencing
-scikit-image==0.21.0
-    # via napari (setup.cfg)
-scipy==1.11.1 ; python_version >= "3.9"
+scikit-image==0.22.0
+    # via napari (napari_repo/setup.cfg)
+scipy==1.11.3 ; python_version >= "3.9"
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   scikit-image
 shiboken2==5.15.2.1
     # via pyside2
@@ -438,44 +434,51 @@ snowballstemmer==2.2.0
     # via sphinx
 sortedcontainers==2.4.0
     # via hypothesis
-sphinx==4.5.0
+sphinx==7.2.6
     # via
-    #   napari (setup.cfg)
     #   numpydoc
-sphinxcontrib-applehelp==1.0.4
+    #   sphinxcontrib-applehelp
+    #   sphinxcontrib-devhelp
+    #   sphinxcontrib-htmlhelp
+    #   sphinxcontrib-qthelp
+    #   sphinxcontrib-serializinghtml
+sphinxcontrib-applehelp==1.0.7
     # via sphinx
-sphinxcontrib-devhelp==1.0.2
+sphinxcontrib-devhelp==1.0.5
     # via sphinx
-sphinxcontrib-htmlhelp==2.0.1
+sphinxcontrib-htmlhelp==2.0.4
     # via sphinx
 sphinxcontrib-jsmath==1.0.1
     # via sphinx
-sphinxcontrib-qthelp==1.0.3
+sphinxcontrib-qthelp==1.0.6
     # via sphinx
-sphinxcontrib-serializinghtml==1.1.5
+sphinxcontrib-serializinghtml==1.1.9
     # via sphinx
-stack-data==0.6.2
+stack-data==0.6.3
     # via ipython
-superqt==0.5.0
+superqt==0.6.0
     # via
     #   magicgui
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   napari-plugin-manager
 sympy==1.12
     # via torch
-tensorstore==0.1.40
+tabulate==0.9.0
+    # via numpydoc
+tensorstore==0.1.45
     # via
-    #   -r resources/constraints/version_denylist.txt
-    #   napari (setup.cfg)
-tifffile==2023.7.18
+    #   -r napari_repo/resources/constraints/version_denylist.txt
+    #   napari (napari_repo/setup.cfg)
+tifffile==2023.9.26
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   scikit-image
 tomli==2.0.1
     # via
     #   build
     #   coverage
     #   npe2
+    #   numpydoc
     #   pyproject-hooks
     #   pytest
 tomli-w==1.0.0
@@ -483,19 +486,19 @@ tomli-w==1.0.0
 toolz==0.12.0
     # via
     #   dask
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   partd
 torch==2.0.1
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   triton
-tornado==6.3.2
+tornado==6.3.3
     # via
     #   ipykernel
     #   jupyter-client
-tqdm==4.65.0
-    # via napari (setup.cfg)
-traitlets==5.9.0
+tqdm==4.66.1
+    # via napari (napari_repo/setup.cfg)
+traitlets==5.11.2
     # via
     #   comm
     #   ipykernel
@@ -508,11 +511,11 @@ triton==2.0.0
     # via torch
 typer==0.9.0
     # via npe2
-typing-extensions==4.7.1
+typing-extensions==4.8.0
     # via
     #   app-model
     #   magicgui
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   pint
     #   psygnal
     #   pydantic
@@ -521,17 +524,17 @@ typing-extensions==4.7.1
     #   typer
 tzdata==2023.3
     # via pandas
-urllib3==2.0.4
+urllib3==2.0.6
     # via requests
-virtualenv==20.24.2
-    # via napari (setup.cfg)
-vispy==0.13.0
+virtualenv==20.24.5
+    # via napari (napari_repo/setup.cfg)
+vispy==0.14.1
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   napari-svg
-wcwidth==0.2.6
+wcwidth==0.2.8
     # via prompt-toolkit
-wheel==0.41.1
+wheel==0.41.2
     # via
     #   nvidia-cublas-cu11
     #   nvidia-cuda-cupti-cu11
@@ -540,18 +543,18 @@ wheel==0.41.1
     #   nvidia-cusparse-cu11
     #   nvidia-nvtx-cu11
 wrapt==1.15.0
-    # via napari (setup.cfg)
-xarray==2023.7.0
-    # via napari (setup.cfg)
-zarr==2.16.0
-    # via napari (setup.cfg)
-zipp==3.16.2
+    # via napari (napari_repo/setup.cfg)
+xarray==2023.9.0
+    # via napari (napari_repo/setup.cfg)
+zarr==2.16.1
+    # via napari (napari_repo/setup.cfg)
+zipp==3.17.0
     # via importlib-metadata

 # The following packages are considered to be unsafe in a requirements file:
 pip==23.2.1
     # via napari-plugin-manager
-setuptools==68.0.0
+setuptools==68.2.2
     # via
     #   nvidia-cublas-cu11
     #   nvidia-cuda-cupti-cu11
diff --git a/resources/constraints/constraints_py3.11.txt b/resources/constraints/constraints_py3.11.txt
index 4bf09eab..d1011933 100644
--- a/resources/constraints/constraints_py3.11.txt
+++ b/resources/constraints/constraints_py3.11.txt
@@ -2,60 +2,60 @@
 # This file is autogenerated by pip-compile with Python 3.11
 # by the following command:
 #
-#    pip-compile --allow-unsafe --extra=pyqt5 --extra=pyqt6_experimental --extra=pyside2 --extra=pyside6_experimental --extra=testing --output-file=resources/constraints/constraints_py3.11.txt --strip-extras resources/constraints/version_denylist.txt setup.cfg
+#    pip-compile --allow-unsafe --extra=pyqt5 --extra=pyqt6_experimental --extra=pyside2 --extra=pyside6_experimental --extra=testing --output-file=napari_repo/resources/constraints/constraints_py3.11.txt --strip-extras napari_repo/resources/constraints/version_denylist.txt napari_repo/setup.cfg
 #
 alabaster==0.7.13
     # via sphinx
-app-model==0.2.0
-    # via napari (setup.cfg)
+app-model==0.2.2
+    # via napari (napari_repo/setup.cfg)
 appdirs==1.4.4
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   npe2
 asciitree==0.3.3
     # via zarr
-asttokens==2.2.1
+asttokens==2.4.0
     # via stack-data
 attrs==23.1.0
     # via
     #   hypothesis
     #   jsonschema
     #   referencing
-babel==2.12.1
+babel==2.13.0
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   sphinx
 backcall==0.2.0
     # via ipython
-build==0.10.0
+build==1.0.3
     # via npe2
 cachey==0.2.1
-    # via napari (setup.cfg)
+    # via napari (napari_repo/setup.cfg)
 certifi==2023.7.22
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   requests
-charset-normalizer==3.2.0
+charset-normalizer==3.3.0
     # via requests
-click==8.1.6
+click==8.1.7
     # via
     #   dask
     #   typer
 cloudpickle==2.2.1
     # via dask
-cmake==3.27.1
+cmake==3.27.6
     # via triton
 comm==0.1.4
     # via ipykernel
-contourpy==1.1.0
+contourpy==1.1.1
     # via matplotlib
-coverage==7.2.7
+coverage==7.3.2
     # via pytest-cov
-cycler==0.11.0
+cycler==0.12.0
     # via matplotlib
-dask==2023.8.0
-    # via napari (setup.cfg)
-debugpy==1.6.7
+dask==2023.9.3
+    # via napari (napari_repo/setup.cfg)
+debugpy==1.8.0
     # via ipykernel
 decorator==5.1.1
     # via ipython
@@ -63,38 +63,38 @@ distlib==0.3.7
     # via virtualenv
 docstring-parser==0.15
     # via magicgui
-docutils==0.17.1
+docutils==0.20.1
     # via sphinx
 entrypoints==0.4
     # via numcodecs
-executing==1.2.0
+executing==2.0.0
     # via stack-data
-fasteners==0.18
+fasteners==0.19
     # via zarr
-filelock==3.12.2
+filelock==3.12.4
     # via
     #   torch
     #   triton
     #   virtualenv
-fonttools==4.42.0
+fonttools==4.43.0
     # via matplotlib
 freetype-py==2.4.0
     # via vispy
-fsspec==2023.6.0
+fsspec==2023.9.2
     # via
     #   dask
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
 heapdict==1.0.1
     # via cachey
-hsluv==5.0.3
+hsluv==5.0.4
     # via vispy
-hypothesis==6.82.2
-    # via napari (setup.cfg)
+hypothesis==6.87.1
+    # via napari (napari_repo/setup.cfg)
 idna==3.4
     # via requests
-imageio==2.31.1
+imageio==2.31.5
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   napari-svg
     #   scikit-image
 imagesize==1.4.1
@@ -105,59 +105,59 @@ in-n-out==0.1.8
     # via app-model
 iniconfig==2.0.0
     # via pytest
-ipykernel==6.25.0
+ipykernel==6.25.2
     # via
     #   napari-console
     #   qtconsole
-ipython==8.14.0
+ipython==8.16.1
     # via
     #   ipykernel
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   napari-console
 ipython-genutils==0.2.0
     # via qtconsole
-jedi==0.19.0
+jedi==0.19.1
     # via ipython
 jinja2==3.1.2
     # via
     #   numpydoc
     #   sphinx
     #   torch
-jsonschema==4.19.0
-    # via napari (setup.cfg)
+jsonschema==4.19.1
+    # via napari (napari_repo/setup.cfg)
 jsonschema-specifications==2023.7.1
     # via jsonschema
-jupyter-client==8.3.0
+jupyter-client==8.3.1
     # via
     #   ipykernel
     #   qtconsole
-jupyter-core==5.3.1
+jupyter-core==5.3.2
     # via
     #   ipykernel
     #   jupyter-client
     #   qtconsole
-kiwisolver==1.4.4
+kiwisolver==1.4.5
     # via
     #   matplotlib
     #   vispy
 lazy-loader==0.3
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   scikit-image
-lit==16.0.6
+lit==17.0.2
     # via triton
 locket==1.0.0
     # via partd
 lxml==4.9.3
-    # via napari (setup.cfg)
-magicgui==0.7.2
-    # via napari (setup.cfg)
+    # via napari (napari_repo/setup.cfg)
+magicgui==0.7.3
+    # via napari (napari_repo/setup.cfg)
 markdown-it-py==3.0.0
     # via rich
 markupsafe==2.1.3
     # via jinja2
-matplotlib==3.7.2
-    # via napari (setup.cfg)
+matplotlib==3.8.0
+    # via napari (napari_repo/setup.cfg)
 matplotlib-inline==0.1.6
     # via
     #   ipykernel
@@ -169,17 +169,17 @@ mpmath==1.3.0
 mypy-extensions==1.0.0
     # via psygnal
 napari-console==0.0.8
-    # via napari (setup.cfg)
+    # via napari (napari_repo/setup.cfg)
 napari-plugin-engine==0.2.0
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   napari-console
     #   napari-svg
 napari-plugin-manager==0.1.0a2
-    # via napari (setup.cfg)
+    # via napari (napari_repo/setup.cfg)
 napari-svg==0.1.10
-    # via napari (setup.cfg)
-nest-asyncio==1.5.7
+    # via napari (napari_repo/setup.cfg)
+nest-asyncio==1.5.8
     # via ipykernel
 networkx==3.1
     # via
@@ -187,21 +187,20 @@ networkx==3.1
     #   torch
 npe2==0.7.2
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   napari-plugin-manager
 numcodecs==0.11.0
     # via zarr
-numpy==1.25.2
+numpy==1.26.0
     # via
     #   contourpy
     #   dask
     #   imageio
     #   matplotlib
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   napari-svg
     #   numcodecs
     #   pandas
-    #   pywavelets
     #   scikit-image
     #   scipy
     #   tensorstore
@@ -209,8 +208,8 @@ numpy==1.25.2
     #   vispy
     #   xarray
     #   zarr
-numpydoc==1.5.0
-    # via napari (setup.cfg)
+numpydoc==1.6.0
+    # via napari (napari_repo/setup.cfg)
 nvidia-cublas-cu11==11.10.3.66
     # via
     #   nvidia-cudnn-cu11
@@ -236,7 +235,7 @@ nvidia-nccl-cu11==2.14.3
     # via torch
 nvidia-nvtx-cu11==11.7.91
     # via torch
-packaging==23.1
+packaging==23.2
     # via
     #   build
     #   dask
@@ -251,101 +250,101 @@ packaging==23.1
     #   superqt
     #   vispy
     #   xarray
-pandas==2.0.3 ; python_version >= "3.9"
+pandas==2.1.1 ; python_version >= "3.9"
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   xarray
 parso==0.8.3
     # via jedi
-partd==1.4.0
+partd==1.4.1
     # via dask
 pexpect==4.8.0
     # via ipython
 pickleshare==0.7.5
     # via ipython
-pillow==10.0.0
+pillow==10.0.1
     # via
     #   imageio
     #   matplotlib
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   scikit-image
 pint==0.22
-    # via napari (setup.cfg)
-platformdirs==3.10.0
+    # via napari (napari_repo/setup.cfg)
+platformdirs==3.11.0
     # via
     #   jupyter-core
     #   pooch
     #   virtualenv
-pluggy==1.2.0
+pluggy==1.3.0
     # via pytest
 pooch==1.7.0
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   scikit-image
 prompt-toolkit==3.0.39
     # via ipython
 psutil==5.9.5
     # via
     #   ipykernel
-    #   napari (setup.cfg)
-psygnal==0.9.1
+    #   napari (napari_repo/setup.cfg)
+psygnal==0.9.4
     # via
     #   app-model
     #   magicgui
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   npe2
 ptyprocess==0.7.0
     # via pexpect
 pure-eval==0.2.2
     # via stack-data
-pydantic==1.10.12
+pydantic==1.10.13
     # via
     #   app-model
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   npe2
 pygments==2.16.1
     # via
     #   ipython
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   qtconsole
     #   rich
     #   sphinx
     #   superqt
 pyopengl==3.1.6
     # via
-    #   -r resources/constraints/version_denylist.txt
-    #   napari (setup.cfg)
-pyparsing==3.0.9
+    #   -r napari_repo/resources/constraints/version_denylist.txt
+    #   napari (napari_repo/setup.cfg)
+pyparsing==3.1.1
     # via matplotlib
 pyproject-hooks==1.0.0
     # via build
 pyqt5==5.15.9
-    # via napari (setup.cfg)
+    # via napari (napari_repo/setup.cfg)
 pyqt5-qt5==5.15.2
     # via pyqt5
 pyqt5-sip==12.12.2
     # via pyqt5
 pyqt6==6.5.2
-    # via napari (setup.cfg)
+    # via napari (napari_repo/setup.cfg)
 pyqt6-qt6==6.5.2
     # via pyqt6
 pyqt6-sip==13.5.2
     # via pyqt6
 pyside2==5.13.2 ; python_version != "3.8"
-    # via napari (setup.cfg)
+    # via napari (napari_repo/setup.cfg)
 pyside6==6.4.2 ; python_version >= "3.10"
     # via
-    #   -r resources/constraints/version_denylist.txt
-    #   napari (setup.cfg)
+    #   -r napari_repo/resources/constraints/version_denylist.txt
+    #   napari (napari_repo/setup.cfg)
 pyside6-addons==6.4.2
     # via pyside6
 pyside6-essentials==6.4.2
     # via
     #   pyside6
     #   pyside6-addons
-pytest==7.4.0
+pytest==7.4.2
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   pytest-cov
     #   pytest-json-report
     #   pytest-metadata
@@ -353,43 +352,41 @@ pytest==7.4.0
     #   pytest-qt
 pytest-cov==4.1.0
     # via
-    #   -r resources/constraints/version_denylist.txt
-    #   napari (setup.cfg)
+    #   -r napari_repo/resources/constraints/version_denylist.txt
+    #   napari (napari_repo/setup.cfg)
 pytest-json-report==1.5.0
-    # via -r resources/constraints/version_denylist.txt
+    # via -r napari_repo/resources/constraints/version_denylist.txt
 pytest-metadata==3.0.0
     # via pytest-json-report
 pytest-pretty==1.2.0
-    # via napari (setup.cfg)
+    # via napari (napari_repo/setup.cfg)
 pytest-qt==4.2.0
-    # via napari (setup.cfg)
+    # via napari (napari_repo/setup.cfg)
 python-dateutil==2.8.2
     # via
     #   jupyter-client
     #   matplotlib
     #   pandas
-pytz==2023.3
+pytz==2023.3.post1
     # via pandas
-pywavelets==1.4.1
-    # via scikit-image
 pyyaml==6.0.1
     # via
     #   dask
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   npe2
-pyzmq==25.1.0
+pyzmq==25.1.1
     # via
     #   ipykernel
     #   jupyter-client
     #   qtconsole
-qtconsole==5.4.3
+qtconsole==5.4.4
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   napari-console
-qtpy==2.3.1
+qtpy==2.4.0
     # via
     #   magicgui
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   napari-console
     #   napari-plugin-manager
     #   qtconsole
@@ -402,20 +399,20 @@ requests==2.31.0
     # via
     #   pooch
     #   sphinx
-rich==13.5.2
+rich==13.6.0
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   npe2
     #   pytest-pretty
-rpds-py==0.9.2
+rpds-py==0.10.3
     # via
     #   jsonschema
     #   referencing
-scikit-image==0.21.0
-    # via napari (setup.cfg)
-scipy==1.11.1 ; python_version >= "3.9"
+scikit-image==0.22.0
+    # via napari (napari_repo/setup.cfg)
+scipy==1.11.3 ; python_version >= "3.9"
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   scikit-image
 shiboken2==5.13.2
     # via pyside2
@@ -432,57 +429,63 @@ snowballstemmer==2.2.0
     # via sphinx
 sortedcontainers==2.4.0
     # via hypothesis
-sphinx==4.5.0
+sphinx==7.2.6
     # via
-    #   napari (setup.cfg)
     #   numpydoc
-sphinxcontrib-applehelp==1.0.4
+    #   sphinxcontrib-applehelp
+    #   sphinxcontrib-devhelp
+    #   sphinxcontrib-htmlhelp
+    #   sphinxcontrib-qthelp
+    #   sphinxcontrib-serializinghtml
+sphinxcontrib-applehelp==1.0.7
     # via sphinx
-sphinxcontrib-devhelp==1.0.2
+sphinxcontrib-devhelp==1.0.5
     # via sphinx
-sphinxcontrib-htmlhelp==2.0.1
+sphinxcontrib-htmlhelp==2.0.4
     # via sphinx
 sphinxcontrib-jsmath==1.0.1
     # via sphinx
-sphinxcontrib-qthelp==1.0.3
+sphinxcontrib-qthelp==1.0.6
     # via sphinx
-sphinxcontrib-serializinghtml==1.1.5
+sphinxcontrib-serializinghtml==1.1.9
     # via sphinx
-stack-data==0.6.2
+stack-data==0.6.3
     # via ipython
-superqt==0.5.0
+superqt==0.6.0
     # via
     #   magicgui
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   napari-plugin-manager
 sympy==1.12
     # via torch
-tensorstore==0.1.40
+tabulate==0.9.0
+    # via numpydoc
+tensorstore==0.1.45
     # via
-    #   -r resources/constraints/version_denylist.txt
-    #   napari (setup.cfg)
-tifffile==2023.7.18
+    #   -r napari_repo/resources/constraints/version_denylist.txt
+    #   napari (napari_repo/setup.cfg)
+tifffile==2023.9.26
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   scikit-image
 tomli-w==1.0.0
     # via npe2
 toolz==0.12.0
     # via
     #   dask
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   partd
 torch==2.0.1
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   triton
-tornado==6.3.2
+tornado==6.3.3
     # via
     #   ipykernel
     #   jupyter-client
-tqdm==4.65.0
-    # via napari (setup.cfg)
-traitlets==5.9.0
+tqdm==4.66.1
+    # via napari (napari_repo/setup.cfg)
+traitlets==5.11.2
     # via
     #   comm
     #   ipykernel
@@ -495,11 +498,11 @@ triton==2.0.0
     # via torch
 typer==0.9.0
     # via npe2
-typing-extensions==4.7.1
+typing-extensions==4.8.0
     # via
     #   app-model
     #   magicgui
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   pint
     #   psygnal
     #   pydantic
@@ -508,17 +511,17 @@ typing-extensions==4.7.1
     #   typer
 tzdata==2023.3
     # via pandas
-urllib3==2.0.4
+urllib3==2.0.6
     # via requests
-virtualenv==20.24.2
-    # via napari (setup.cfg)
-vispy==0.13.0
+virtualenv==20.24.5
+    # via napari (napari_repo/setup.cfg)
+vispy==0.14.1
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   napari-svg
-wcwidth==0.2.6
+wcwidth==0.2.8
     # via prompt-toolkit
-wheel==0.41.1
+wheel==0.41.2
     # via
     #   nvidia-cublas-cu11
     #   nvidia-cuda-cupti-cu11
@@ -527,18 +530,18 @@ wheel==0.41.1
     #   nvidia-cusparse-cu11
     #   nvidia-nvtx-cu11
 wrapt==1.15.0
-    # via napari (setup.cfg)
-xarray==2023.7.0
-    # via napari (setup.cfg)
-zarr==2.16.0
-    # via napari (setup.cfg)
-zipp==3.16.2
+    # via napari (napari_repo/setup.cfg)
+xarray==2023.9.0
+    # via napari (napari_repo/setup.cfg)
+zarr==2.16.1
+    # via napari (napari_repo/setup.cfg)
+zipp==3.17.0
     # via importlib-metadata

 # The following packages are considered to be unsafe in a requirements file:
 pip==23.2.1
     # via napari-plugin-manager
-setuptools==68.0.0
+setuptools==68.2.2
     # via
     #   nvidia-cublas-cu11
     #   nvidia-cuda-cupti-cu11
diff --git a/resources/constraints/constraints_py3.8.txt b/resources/constraints/constraints_py3.8.txt
index 99a6f863..a8724acd 100644
--- a/resources/constraints/constraints_py3.8.txt
+++ b/resources/constraints/constraints_py3.8.txt
@@ -2,60 +2,60 @@
 # This file is autogenerated by pip-compile with Python 3.8
 # by the following command:
 #
-#    pip-compile --allow-unsafe --extra=pyqt5 --extra=pyqt6_experimental --extra=pyside2 --extra=pyside6_experimental --extra=testing --output-file=resources/constraints/constraints_py3.8.txt --strip-extras resources/constraints/version_denylist.txt setup.cfg
+#    pip-compile --allow-unsafe --extra=pyqt5 --extra=pyqt6_experimental --extra=pyside2 --extra=pyside6_experimental --extra=testing --output-file=napari_repo/resources/constraints/constraints_py3.8.txt --strip-extras napari_repo/resources/constraints/version_denylist.txt napari_repo/setup.cfg
 #
 alabaster==0.7.13
     # via sphinx
-app-model==0.2.0
-    # via napari (setup.cfg)
+app-model==0.2.2
+    # via napari (napari_repo/setup.cfg)
 appdirs==1.4.4
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   npe2
 asciitree==0.3.3
     # via zarr
-asttokens==2.2.1
+asttokens==2.4.0
     # via stack-data
 attrs==23.1.0
     # via
     #   hypothesis
     #   jsonschema
     #   referencing
-babel==2.12.1
+babel==2.13.0
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   sphinx
 backcall==0.2.0
     # via ipython
-build==0.10.0
+build==1.0.3
     # via npe2
 cachey==0.2.1
-    # via napari (setup.cfg)
+    # via napari (napari_repo/setup.cfg)
 certifi==2023.7.22
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   requests
-charset-normalizer==3.2.0
+charset-normalizer==3.3.0
     # via requests
-click==8.1.6
+click==8.1.7
     # via
     #   dask
     #   typer
 cloudpickle==2.2.1
     # via dask
-cmake==3.27.1
+cmake==3.27.6
     # via triton
 comm==0.1.4
     # via ipykernel
-contourpy==1.1.0
+contourpy==1.1.1
     # via matplotlib
-coverage==7.2.7
+coverage==7.3.2
     # via pytest-cov
-cycler==0.11.0
+cycler==0.12.0
     # via matplotlib
 dask==2023.5.0
-    # via napari (setup.cfg)
-debugpy==1.6.7
+    # via napari (napari_repo/setup.cfg)
+debugpy==1.8.0
     # via ipykernel
 decorator==5.1.1
     # via ipython
@@ -63,52 +63,53 @@ distlib==0.3.7
     # via virtualenv
 docstring-parser==0.15
     # via magicgui
-docutils==0.17.1
+docutils==0.20.1
     # via sphinx
 entrypoints==0.4
     # via numcodecs
-exceptiongroup==1.1.2
+exceptiongroup==1.1.3
     # via
     #   hypothesis
     #   pytest
-executing==1.2.0
+executing==2.0.0
     # via stack-data
-fasteners==0.18
+fasteners==0.19
     # via zarr
-filelock==3.12.2
+filelock==3.12.4
     # via
     #   torch
     #   triton
     #   virtualenv
-fonttools==4.42.0
+fonttools==4.43.0
     # via matplotlib
 freetype-py==2.4.0
     # via vispy
-fsspec==2023.6.0
+fsspec==2023.9.2
     # via
     #   dask
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
 heapdict==1.0.1
     # via cachey
-hsluv==5.0.3
+hsluv==5.0.4
     # via vispy
-hypothesis==6.82.2
-    # via napari (setup.cfg)
+hypothesis==6.87.1
+    # via napari (napari_repo/setup.cfg)
 idna==3.4
     # via requests
-imageio==2.31.1
+imageio==2.31.5
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   napari-svg
     #   scikit-image
 imagesize==1.4.1
     # via sphinx
 importlib-metadata==6.8.0
     # via
+    #   build
     #   dask
     #   jupyter-client
     #   sphinx
-importlib-resources==6.0.1
+importlib-resources==6.1.0
     # via
     #   jsonschema
     #   jsonschema-specifications
@@ -117,59 +118,59 @@ in-n-out==0.1.8
     # via app-model
 iniconfig==2.0.0
     # via pytest
-ipykernel==6.25.0
+ipykernel==6.25.2
     # via
     #   napari-console
     #   qtconsole
-ipython==8.12.2
+ipython==8.12.3
     # via
     #   ipykernel
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   napari-console
 ipython-genutils==0.2.0
     # via qtconsole
-jedi==0.19.0
+jedi==0.19.1
     # via ipython
 jinja2==3.1.2
     # via
     #   numpydoc
     #   sphinx
     #   torch
-jsonschema==4.19.0
-    # via napari (setup.cfg)
+jsonschema==4.19.1
+    # via napari (napari_repo/setup.cfg)
 jsonschema-specifications==2023.7.1
     # via jsonschema
-jupyter-client==8.3.0
+jupyter-client==8.3.1
     # via
     #   ipykernel
     #   qtconsole
-jupyter-core==5.3.1
+jupyter-core==5.3.2
     # via
     #   ipykernel
     #   jupyter-client
     #   qtconsole
-kiwisolver==1.4.4
+kiwisolver==1.4.5
     # via
     #   matplotlib
     #   vispy
 lazy-loader==0.3
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   scikit-image
-lit==16.0.6
+lit==17.0.2
     # via triton
 locket==1.0.0
     # via partd
 lxml==4.9.3
-    # via napari (setup.cfg)
-magicgui==0.7.2
-    # via napari (setup.cfg)
+    # via napari (napari_repo/setup.cfg)
+magicgui==0.7.3
+    # via napari (napari_repo/setup.cfg)
 markdown-it-py==3.0.0
     # via rich
 markupsafe==2.1.3
     # via jinja2
-matplotlib==3.7.2
-    # via napari (setup.cfg)
+matplotlib==3.7.3
+    # via napari (napari_repo/setup.cfg)
 matplotlib-inline==0.1.6
     # via
     #   ipykernel
@@ -181,17 +182,17 @@ mpmath==1.3.0
 mypy-extensions==1.0.0
     # via psygnal
 napari-console==0.0.8
-    # via napari (setup.cfg)
+    # via napari (napari_repo/setup.cfg)
 napari-plugin-engine==0.2.0
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   napari-console
     #   napari-svg
 napari-plugin-manager==0.1.0a2
-    # via napari (setup.cfg)
+    # via napari (napari_repo/setup.cfg)
 napari-svg==0.1.10
-    # via napari (setup.cfg)
-nest-asyncio==1.5.7
+    # via napari (napari_repo/setup.cfg)
+nest-asyncio==1.5.8
     # via ipykernel
 networkx==3.1
     # via
@@ -199,7 +200,7 @@ networkx==3.1
     #   torch
 npe2==0.7.2
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   napari-plugin-manager
 numcodecs==0.11.0
     # via zarr
@@ -209,7 +210,7 @@ numpy==1.24.4
     #   dask
     #   imageio
     #   matplotlib
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   napari-svg
     #   numcodecs
     #   pandas
@@ -221,8 +222,8 @@ numpy==1.24.4
     #   vispy
     #   xarray
     #   zarr
-numpydoc==1.5.0
-    # via napari (setup.cfg)
+numpydoc==1.6.0
+    # via napari (napari_repo/setup.cfg)
 nvidia-cublas-cu11==11.10.3.66
     # via
     #   nvidia-cudnn-cu11
@@ -248,7 +249,7 @@ nvidia-nccl-cu11==2.14.3
     # via torch
 nvidia-nvtx-cu11==11.7.91
     # via torch
-packaging==23.1
+packaging==23.2
     # via
     #   build
     #   dask
@@ -265,101 +266,101 @@ packaging==23.1
     #   xarray
 pandas==2.0.3 ; python_version < "3.9"
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   xarray
 parso==0.8.3
     # via jedi
-partd==1.4.0
+partd==1.4.1
     # via dask
 pexpect==4.8.0
     # via ipython
 pickleshare==0.7.5
     # via ipython
-pillow==10.0.0
+pillow==10.0.1
     # via
     #   imageio
     #   matplotlib
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   scikit-image
 pint==0.21.1
-    # via napari (setup.cfg)
+    # via napari (napari_repo/setup.cfg)
 pkgutil-resolve-name==1.3.10
     # via jsonschema
-platformdirs==3.10.0
+platformdirs==3.11.0
     # via
     #   jupyter-core
     #   pooch
     #   virtualenv
-pluggy==1.2.0
+pluggy==1.3.0
     # via pytest
 pooch==1.7.0
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   scikit-image
 prompt-toolkit==3.0.39
     # via ipython
 psutil==5.9.5
     # via
     #   ipykernel
-    #   napari (setup.cfg)
-psygnal==0.9.1
+    #   napari (napari_repo/setup.cfg)
+psygnal==0.9.4
     # via
     #   app-model
     #   magicgui
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   npe2
 ptyprocess==0.7.0
     # via pexpect
 pure-eval==0.2.2
     # via stack-data
-pydantic==1.10.12
+pydantic==1.10.13
     # via
     #   app-model
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   npe2
 pygments==2.16.1
     # via
     #   ipython
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   qtconsole
     #   rich
     #   sphinx
     #   superqt
 pyopengl==3.1.6
     # via
-    #   -r resources/constraints/version_denylist.txt
-    #   napari (setup.cfg)
-pyparsing==3.0.9
+    #   -r napari_repo/resources/constraints/version_denylist.txt
+    #   napari (napari_repo/setup.cfg)
+pyparsing==3.1.1
     # via matplotlib
 pyproject-hooks==1.0.0
     # via build
 pyqt5==5.15.9
-    # via napari (setup.cfg)
+    # via napari (napari_repo/setup.cfg)
 pyqt5-qt5==5.15.2
     # via pyqt5
 pyqt5-sip==12.12.2
     # via pyqt5
 pyqt6==6.5.2
-    # via napari (setup.cfg)
+    # via napari (napari_repo/setup.cfg)
 pyqt6-qt6==6.5.2
     # via pyqt6
 pyqt6-sip==13.5.2
     # via pyqt6
 pyside2==5.15.2.1 ; python_version == "3.8"
-    # via napari (setup.cfg)
+    # via napari (napari_repo/setup.cfg)
 pyside6==6.3.1 ; python_version < "3.10"
     # via
-    #   -r resources/constraints/version_denylist.txt
-    #   napari (setup.cfg)
+    #   -r napari_repo/resources/constraints/version_denylist.txt
+    #   napari (napari_repo/setup.cfg)
 pyside6-addons==6.3.1
     # via pyside6
 pyside6-essentials==6.3.1
     # via
     #   pyside6
     #   pyside6-addons
-pytest==7.4.0
+pytest==7.4.2
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   pytest-cov
     #   pytest-json-report
     #   pytest-metadata
@@ -367,22 +368,22 @@ pytest==7.4.0
     #   pytest-qt
 pytest-cov==4.1.0
     # via
-    #   -r resources/constraints/version_denylist.txt
-    #   napari (setup.cfg)
+    #   -r napari_repo/resources/constraints/version_denylist.txt
+    #   napari (napari_repo/setup.cfg)
 pytest-json-report==1.5.0
-    # via -r resources/constraints/version_denylist.txt
+    # via -r napari_repo/resources/constraints/version_denylist.txt
 pytest-metadata==3.0.0
     # via pytest-json-report
 pytest-pretty==1.2.0
-    # via napari (setup.cfg)
+    # via napari (napari_repo/setup.cfg)
 pytest-qt==4.2.0
-    # via napari (setup.cfg)
+    # via napari (napari_repo/setup.cfg)
 python-dateutil==2.8.2
     # via
     #   jupyter-client
     #   matplotlib
     #   pandas
-pytz==2023.3
+pytz==2023.3.post1
     # via
     #   babel
     #   pandas
@@ -391,21 +392,21 @@ pywavelets==1.4.1
 pyyaml==6.0.1
     # via
     #   dask
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   npe2
-pyzmq==25.1.0
+pyzmq==25.1.1
     # via
     #   ipykernel
     #   jupyter-client
     #   qtconsole
-qtconsole==5.4.3
+qtconsole==5.4.4
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   napari-console
-qtpy==2.3.1
+qtpy==2.4.0
     # via
     #   magicgui
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   napari-console
     #   napari-plugin-manager
     #   qtconsole
@@ -418,20 +419,20 @@ requests==2.31.0
     # via
     #   pooch
     #   sphinx
-rich==13.5.2
+rich==13.6.0
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   npe2
     #   pytest-pretty
-rpds-py==0.9.2
+rpds-py==0.10.3
     # via
     #   jsonschema
     #   referencing
 scikit-image==0.21.0
-    # via napari (setup.cfg)
+    # via napari (napari_repo/setup.cfg)
 scipy==1.10.1 ; python_version < "3.9"
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   scikit-image
 shiboken2==5.15.2.1
     # via pyside2
@@ -448,10 +449,8 @@ snowballstemmer==2.2.0
     # via sphinx
 sortedcontainers==2.4.0
     # via hypothesis
-sphinx==4.5.0
-    # via
-    #   napari (setup.cfg)
-    #   numpydoc
+sphinx==7.1.2
+    # via numpydoc
 sphinxcontrib-applehelp==1.0.4
     # via sphinx
 sphinxcontrib-devhelp==1.0.2
@@ -464,28 +463,31 @@ sphinxcontrib-qthelp==1.0.3
     # via sphinx
 sphinxcontrib-serializinghtml==1.1.5
     # via sphinx
-stack-data==0.6.2
+stack-data==0.6.3
     # via ipython
-superqt==0.5.0
+superqt==0.6.0
     # via
     #   magicgui
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   napari-plugin-manager
 sympy==1.12
     # via torch
-tensorstore==0.1.40
+tabulate==0.9.0
+    # via numpydoc
+tensorstore==0.1.45
     # via
-    #   -r resources/constraints/version_denylist.txt
-    #   napari (setup.cfg)
+    #   -r napari_repo/resources/constraints/version_denylist.txt
+    #   napari (napari_repo/setup.cfg)
 tifffile==2023.7.10
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   scikit-image
 tomli==2.0.1
     # via
     #   build
     #   coverage
     #   npe2
+    #   numpydoc
     #   pyproject-hooks
     #   pytest
 tomli-w==1.0.0
@@ -493,19 +495,19 @@ tomli-w==1.0.0
 toolz==0.12.0
     # via
     #   dask
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   partd
 torch==2.0.1
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   triton
-tornado==6.3.2
+tornado==6.3.3
     # via
     #   ipykernel
     #   jupyter-client
-tqdm==4.65.0
-    # via napari (setup.cfg)
-traitlets==5.9.0
+tqdm==4.66.1
+    # via napari (napari_repo/setup.cfg)
+traitlets==5.11.2
     # via
     #   comm
     #   ipykernel
@@ -518,12 +520,12 @@ triton==2.0.0
     # via torch
 typer==0.9.0
     # via npe2
-typing-extensions==4.7.1
+typing-extensions==4.8.0
     # via
     #   app-model
     #   ipython
     #   magicgui
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   psygnal
     #   pydantic
     #   rich
@@ -532,17 +534,17 @@ typing-extensions==4.7.1
     #   typer
 tzdata==2023.3
     # via pandas
-urllib3==2.0.4
+urllib3==2.0.6
     # via requests
-virtualenv==20.24.2
-    # via napari (setup.cfg)
-vispy==0.13.0
+virtualenv==20.24.5
+    # via napari (napari_repo/setup.cfg)
+vispy==0.14.1
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   napari-svg
-wcwidth==0.2.6
+wcwidth==0.2.8
     # via prompt-toolkit
-wheel==0.41.1
+wheel==0.41.2
     # via
     #   nvidia-cublas-cu11
     #   nvidia-cuda-cupti-cu11
@@ -551,12 +553,12 @@ wheel==0.41.1
     #   nvidia-cusparse-cu11
     #   nvidia-nvtx-cu11
 wrapt==1.15.0
-    # via napari (setup.cfg)
+    # via napari (napari_repo/setup.cfg)
 xarray==2023.1.0
-    # via napari (setup.cfg)
-zarr==2.16.0
-    # via napari (setup.cfg)
-zipp==3.16.2
+    # via napari (napari_repo/setup.cfg)
+zarr==2.16.1
+    # via napari (napari_repo/setup.cfg)
+zipp==3.17.0
     # via
     #   importlib-metadata
     #   importlib-resources
@@ -564,7 +566,7 @@ zipp==3.16.2
 # The following packages are considered to be unsafe in a requirements file:
 pip==23.2.1
     # via napari-plugin-manager
-setuptools==68.0.0
+setuptools==68.2.2
     # via
     #   nvidia-cublas-cu11
     #   nvidia-cuda-cupti-cu11
diff --git a/resources/constraints/constraints_py3.9.txt b/resources/constraints/constraints_py3.9.txt
index b8696e02..ea8f7e74 100644
--- a/resources/constraints/constraints_py3.9.txt
+++ b/resources/constraints/constraints_py3.9.txt
@@ -2,60 +2,60 @@
 # This file is autogenerated by pip-compile with Python 3.9
 # by the following command:
 #
-#    pip-compile --allow-unsafe --extra=pyqt5 --extra=pyqt6_experimental --extra=pyside2 --extra=pyside6_experimental --extra=testing --output-file=resources/constraints/constraints_py3.9.txt --strip-extras resources/constraints/version_denylist.txt setup.cfg
+#    pip-compile --allow-unsafe --extra=pyqt5 --extra=pyqt6_experimental --extra=pyside2 --extra=pyside6_experimental --extra=testing --output-file=napari_repo/resources/constraints/constraints_py3.9.txt --strip-extras napari_repo/resources/constraints/version_denylist.txt napari_repo/setup.cfg
 #
 alabaster==0.7.13
     # via sphinx
-app-model==0.2.0
-    # via napari (setup.cfg)
+app-model==0.2.2
+    # via napari (napari_repo/setup.cfg)
 appdirs==1.4.4
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   npe2
 asciitree==0.3.3
     # via zarr
-asttokens==2.2.1
+asttokens==2.4.0
     # via stack-data
 attrs==23.1.0
     # via
     #   hypothesis
     #   jsonschema
     #   referencing
-babel==2.12.1
+babel==2.13.0
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   sphinx
 backcall==0.2.0
     # via ipython
-build==0.10.0
+build==1.0.3
     # via npe2
 cachey==0.2.1
-    # via napari (setup.cfg)
+    # via napari (napari_repo/setup.cfg)
 certifi==2023.7.22
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   requests
-charset-normalizer==3.2.0
+charset-normalizer==3.3.0
     # via requests
-click==8.1.6
+click==8.1.7
     # via
     #   dask
     #   typer
 cloudpickle==2.2.1
     # via dask
-cmake==3.27.1
+cmake==3.27.6
     # via triton
 comm==0.1.4
     # via ipykernel
-contourpy==1.1.0
+contourpy==1.1.1
     # via matplotlib
-coverage==7.2.7
+coverage==7.3.2
     # via pytest-cov
-cycler==0.11.0
+cycler==0.12.0
     # via matplotlib
-dask==2023.8.0
-    # via napari (setup.cfg)
-debugpy==1.6.7
+dask==2023.9.3
+    # via napari (napari_repo/setup.cfg)
+debugpy==1.8.0
     # via ipykernel
 decorator==5.1.1
     # via ipython
@@ -63,110 +63,112 @@ distlib==0.3.7
     # via virtualenv
 docstring-parser==0.15
     # via magicgui
-docutils==0.17.1
+docutils==0.20.1
     # via sphinx
 entrypoints==0.4
     # via numcodecs
-exceptiongroup==1.1.2
+exceptiongroup==1.1.3
     # via
     #   hypothesis
+    #   ipython
     #   pytest
-executing==1.2.0
+executing==2.0.0
     # via stack-data
-fasteners==0.18
+fasteners==0.19
     # via zarr
-filelock==3.12.2
+filelock==3.12.4
     # via
     #   torch
     #   triton
     #   virtualenv
-fonttools==4.42.0
+fonttools==4.43.0
     # via matplotlib
 freetype-py==2.4.0
     # via vispy
-fsspec==2023.6.0
+fsspec==2023.9.2
     # via
     #   dask
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
 heapdict==1.0.1
     # via cachey
-hsluv==5.0.3
+hsluv==5.0.4
     # via vispy
-hypothesis==6.82.2
-    # via napari (setup.cfg)
+hypothesis==6.87.1
+    # via napari (napari_repo/setup.cfg)
 idna==3.4
     # via requests
-imageio==2.31.1
+imageio==2.31.5
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   napari-svg
     #   scikit-image
 imagesize==1.4.1
     # via sphinx
 importlib-metadata==6.8.0
     # via
+    #   build
     #   dask
     #   jupyter-client
     #   sphinx
-importlib-resources==6.0.1
+importlib-resources==6.1.0
     # via matplotlib
 in-n-out==0.1.8
     # via app-model
 iniconfig==2.0.0
     # via pytest
-ipykernel==6.25.0
+ipykernel==6.25.2
     # via
     #   napari-console
     #   qtconsole
-ipython==8.14.0
+ipython==8.16.1
     # via
     #   ipykernel
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   napari-console
 ipython-genutils==0.2.0
     # via qtconsole
-jedi==0.19.0
+jedi==0.19.1
     # via ipython
 jinja2==3.1.2
     # via
     #   numpydoc
     #   sphinx
     #   torch
-jsonschema==4.19.0
-    # via napari (setup.cfg)
+jsonschema==4.19.1
+    # via napari (napari_repo/setup.cfg)
 jsonschema-specifications==2023.7.1
     # via jsonschema
-jupyter-client==8.3.0
+jupyter-client==8.3.1
     # via
     #   ipykernel
     #   qtconsole
-jupyter-core==5.3.1
+jupyter-core==5.3.2
     # via
     #   ipykernel
     #   jupyter-client
     #   qtconsole
-kiwisolver==1.4.4
+kiwisolver==1.4.5
     # via
     #   matplotlib
     #   vispy
 lazy-loader==0.3
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   scikit-image
-lit==16.0.6
+lit==17.0.2
     # via triton
 locket==1.0.0
     # via partd
 lxml==4.9.3
-    # via napari (setup.cfg)
-magicgui==0.7.2
-    # via napari (setup.cfg)
+    # via napari (napari_repo/setup.cfg)
+magicgui==0.7.3
+    # via napari (napari_repo/setup.cfg)
 markdown-it-py==3.0.0
     # via rich
 markupsafe==2.1.3
     # via jinja2
-matplotlib==3.7.2
-    # via napari (setup.cfg)
+matplotlib==3.8.0
+    # via napari (napari_repo/setup.cfg)
 matplotlib-inline==0.1.6
     # via
     #   ipykernel
@@ -178,17 +180,17 @@ mpmath==1.3.0
 mypy-extensions==1.0.0
     # via psygnal
 napari-console==0.0.8
-    # via napari (setup.cfg)
+    # via napari (napari_repo/setup.cfg)
 napari-plugin-engine==0.2.0
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   napari-console
     #   napari-svg
 napari-plugin-manager==0.1.0a2
-    # via napari (setup.cfg)
+    # via napari (napari_repo/setup.cfg)
 napari-svg==0.1.10
-    # via napari (setup.cfg)
-nest-asyncio==1.5.7
+    # via napari (napari_repo/setup.cfg)
+nest-asyncio==1.5.8
     # via ipykernel
 networkx==3.1
     # via
@@ -196,21 +198,20 @@ networkx==3.1
     #   torch
 npe2==0.7.2
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   napari-plugin-manager
 numcodecs==0.11.0
     # via zarr
-numpy==1.25.2
+numpy==1.26.0
     # via
     #   contourpy
     #   dask
     #   imageio
     #   matplotlib
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   napari-svg
     #   numcodecs
     #   pandas
-    #   pywavelets
     #   scikit-image
     #   scipy
     #   tensorstore
@@ -218,8 +219,8 @@ numpy==1.25.2
     #   vispy
     #   xarray
     #   zarr
-numpydoc==1.5.0
-    # via napari (setup.cfg)
+numpydoc==1.6.0
+    # via napari (napari_repo/setup.cfg)
 nvidia-cublas-cu11==11.10.3.66
     # via
     #   nvidia-cudnn-cu11
@@ -245,7 +246,7 @@ nvidia-nccl-cu11==2.14.3
     # via torch
 nvidia-nvtx-cu11==11.7.91
     # via torch
-packaging==23.1
+packaging==23.2
     # via
     #   build
     #   dask
@@ -260,101 +261,101 @@ packaging==23.1
     #   superqt
     #   vispy
     #   xarray
-pandas==2.0.3 ; python_version >= "3.9"
+pandas==2.1.1 ; python_version >= "3.9"
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   xarray
 parso==0.8.3
     # via jedi
-partd==1.4.0
+partd==1.4.1
     # via dask
 pexpect==4.8.0
     # via ipython
 pickleshare==0.7.5
     # via ipython
-pillow==10.0.0
+pillow==10.0.1
     # via
     #   imageio
     #   matplotlib
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   scikit-image
 pint==0.22
-    # via napari (setup.cfg)
-platformdirs==3.10.0
+    # via napari (napari_repo/setup.cfg)
+platformdirs==3.11.0
     # via
     #   jupyter-core
     #   pooch
     #   virtualenv
-pluggy==1.2.0
+pluggy==1.3.0
     # via pytest
 pooch==1.7.0
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   scikit-image
 prompt-toolkit==3.0.39
     # via ipython
 psutil==5.9.5
     # via
     #   ipykernel
-    #   napari (setup.cfg)
-psygnal==0.9.1
+    #   napari (napari_repo/setup.cfg)
+psygnal==0.9.4
     # via
     #   app-model
     #   magicgui
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   npe2
 ptyprocess==0.7.0
     # via pexpect
 pure-eval==0.2.2
     # via stack-data
-pydantic==1.10.12
+pydantic==1.10.13
     # via
     #   app-model
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   npe2
 pygments==2.16.1
     # via
     #   ipython
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   qtconsole
     #   rich
     #   sphinx
     #   superqt
 pyopengl==3.1.6
     # via
-    #   -r resources/constraints/version_denylist.txt
-    #   napari (setup.cfg)
-pyparsing==3.0.9
+    #   -r napari_repo/resources/constraints/version_denylist.txt
+    #   napari (napari_repo/setup.cfg)
+pyparsing==3.1.1
     # via matplotlib
 pyproject-hooks==1.0.0
     # via build
 pyqt5==5.15.9
-    # via napari (setup.cfg)
+    # via napari (napari_repo/setup.cfg)
 pyqt5-qt5==5.15.2
     # via pyqt5
 pyqt5-sip==12.12.2
     # via pyqt5
 pyqt6==6.5.2
-    # via napari (setup.cfg)
+    # via napari (napari_repo/setup.cfg)
 pyqt6-qt6==6.5.2
     # via pyqt6
 pyqt6-sip==13.5.2
     # via pyqt6
 pyside2==5.15.2.1 ; python_version != "3.8"
-    # via napari (setup.cfg)
+    # via napari (napari_repo/setup.cfg)
 pyside6==6.3.1 ; python_version < "3.10"
     # via
-    #   -r resources/constraints/version_denylist.txt
-    #   napari (setup.cfg)
+    #   -r napari_repo/resources/constraints/version_denylist.txt
+    #   napari (napari_repo/setup.cfg)
 pyside6-addons==6.3.1
     # via pyside6
 pyside6-essentials==6.3.1
     # via
     #   pyside6
     #   pyside6-addons
-pytest==7.4.0
+pytest==7.4.2
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   pytest-cov
     #   pytest-json-report
     #   pytest-metadata
@@ -362,43 +363,41 @@ pytest==7.4.0
     #   pytest-qt
 pytest-cov==4.1.0
     # via
-    #   -r resources/constraints/version_denylist.txt
-    #   napari (setup.cfg)
+    #   -r napari_repo/resources/constraints/version_denylist.txt
+    #   napari (napari_repo/setup.cfg)
 pytest-json-report==1.5.0
-    # via -r resources/constraints/version_denylist.txt
+    # via -r napari_repo/resources/constraints/version_denylist.txt
 pytest-metadata==3.0.0
     # via pytest-json-report
 pytest-pretty==1.2.0
-    # via napari (setup.cfg)
+    # via napari (napari_repo/setup.cfg)
 pytest-qt==4.2.0
-    # via napari (setup.cfg)
+    # via napari (napari_repo/setup.cfg)
 python-dateutil==2.8.2
     # via
     #   jupyter-client
     #   matplotlib
     #   pandas
-pytz==2023.3
+pytz==2023.3.post1
     # via pandas
-pywavelets==1.4.1
-    # via scikit-image
 pyyaml==6.0.1
     # via
     #   dask
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   npe2
-pyzmq==25.1.0
+pyzmq==25.1.1
     # via
     #   ipykernel
     #   jupyter-client
     #   qtconsole
-qtconsole==5.4.3
+qtconsole==5.4.4
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   napari-console
-qtpy==2.3.1
+qtpy==2.4.0
     # via
     #   magicgui
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   napari-console
     #   napari-plugin-manager
     #   qtconsole
@@ -411,20 +410,20 @@ requests==2.31.0
     # via
     #   pooch
     #   sphinx
-rich==13.5.2
+rich==13.6.0
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   npe2
     #   pytest-pretty
-rpds-py==0.9.2
+rpds-py==0.10.3
     # via
     #   jsonschema
     #   referencing
-scikit-image==0.21.0
-    # via napari (setup.cfg)
-scipy==1.11.1 ; python_version >= "3.9"
+scikit-image==0.22.0
+    # via napari (napari_repo/setup.cfg)
+scipy==1.11.3 ; python_version >= "3.9"
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   scikit-image
 shiboken2==5.15.2.1
     # via pyside2
@@ -441,44 +440,51 @@ snowballstemmer==2.2.0
     # via sphinx
 sortedcontainers==2.4.0
     # via hypothesis
-sphinx==4.5.0
+sphinx==7.2.6
     # via
-    #   napari (setup.cfg)
     #   numpydoc
-sphinxcontrib-applehelp==1.0.4
+    #   sphinxcontrib-applehelp
+    #   sphinxcontrib-devhelp
+    #   sphinxcontrib-htmlhelp
+    #   sphinxcontrib-qthelp
+    #   sphinxcontrib-serializinghtml
+sphinxcontrib-applehelp==1.0.7
     # via sphinx
-sphinxcontrib-devhelp==1.0.2
+sphinxcontrib-devhelp==1.0.5
     # via sphinx
-sphinxcontrib-htmlhelp==2.0.1
+sphinxcontrib-htmlhelp==2.0.4
     # via sphinx
 sphinxcontrib-jsmath==1.0.1
     # via sphinx
-sphinxcontrib-qthelp==1.0.3
+sphinxcontrib-qthelp==1.0.6
     # via sphinx
-sphinxcontrib-serializinghtml==1.1.5
+sphinxcontrib-serializinghtml==1.1.9
     # via sphinx
-stack-data==0.6.2
+stack-data==0.6.3
     # via ipython
-superqt==0.5.0
+superqt==0.6.0
     # via
     #   magicgui
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   napari-plugin-manager
 sympy==1.12
     # via torch
-tensorstore==0.1.40
+tabulate==0.9.0
+    # via numpydoc
+tensorstore==0.1.45
     # via
-    #   -r resources/constraints/version_denylist.txt
-    #   napari (setup.cfg)
-tifffile==2023.7.18
+    #   -r napari_repo/resources/constraints/version_denylist.txt
+    #   napari (napari_repo/setup.cfg)
+tifffile==2023.9.26
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   scikit-image
 tomli==2.0.1
     # via
     #   build
     #   coverage
     #   npe2
+    #   numpydoc
     #   pyproject-hooks
     #   pytest
 tomli-w==1.0.0
@@ -486,19 +492,19 @@ tomli-w==1.0.0
 toolz==0.12.0
     # via
     #   dask
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   partd
 torch==2.0.1
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   triton
-tornado==6.3.2
+tornado==6.3.3
     # via
     #   ipykernel
     #   jupyter-client
-tqdm==4.65.0
-    # via napari (setup.cfg)
-traitlets==5.9.0
+tqdm==4.66.1
+    # via napari (napari_repo/setup.cfg)
+traitlets==5.11.2
     # via
     #   comm
     #   ipykernel
@@ -511,12 +517,12 @@ triton==2.0.0
     # via torch
 typer==0.9.0
     # via npe2
-typing-extensions==4.7.1
+typing-extensions==4.8.0
     # via
     #   app-model
     #   ipython
     #   magicgui
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   pint
     #   psygnal
     #   pydantic
@@ -525,17 +531,17 @@ typing-extensions==4.7.1
     #   typer
 tzdata==2023.3
     # via pandas
-urllib3==2.0.4
+urllib3==2.0.6
     # via requests
-virtualenv==20.24.2
-    # via napari (setup.cfg)
-vispy==0.13.0
+virtualenv==20.24.5
+    # via napari (napari_repo/setup.cfg)
+vispy==0.14.1
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   napari-svg
-wcwidth==0.2.6
+wcwidth==0.2.8
     # via prompt-toolkit
-wheel==0.41.1
+wheel==0.41.2
     # via
     #   nvidia-cublas-cu11
     #   nvidia-cuda-cupti-cu11
@@ -544,12 +550,12 @@ wheel==0.41.1
     #   nvidia-cusparse-cu11
     #   nvidia-nvtx-cu11
 wrapt==1.15.0
-    # via napari (setup.cfg)
-xarray==2023.7.0
-    # via napari (setup.cfg)
-zarr==2.16.0
-    # via napari (setup.cfg)
-zipp==3.16.2
+    # via napari (napari_repo/setup.cfg)
+xarray==2023.9.0
+    # via napari (napari_repo/setup.cfg)
+zarr==2.16.1
+    # via napari (napari_repo/setup.cfg)
+zipp==3.17.0
     # via
     #   importlib-metadata
     #   importlib-resources
@@ -557,7 +563,7 @@ zipp==3.16.2
 # The following packages are considered to be unsafe in a requirements file:
 pip==23.2.1
     # via napari-plugin-manager
-setuptools==68.0.0
+setuptools==68.2.2
     # via
     #   nvidia-cublas-cu11
     #   nvidia-cuda-cupti-cu11
diff --git a/resources/constraints/constraints_py3.9_examples.txt b/resources/constraints/constraints_py3.9_examples.txt
index 48ae044e..1c79c863 100644
--- a/resources/constraints/constraints_py3.9_examples.txt
+++ b/resources/constraints/constraints_py3.9_examples.txt
@@ -2,60 +2,60 @@
 # This file is autogenerated by pip-compile with Python 3.9
 # by the following command:
 #
-#    pip-compile --allow-unsafe --extra=pyqt5 --extra=pyqt6_experimental --extra=pyside2 --extra=pyside6_experimental --extra=testing --output-file=resources/constraints/constraints_py3.9_examples.txt --strip-extras resources/constraints/version_denylist.txt resources/constraints/version_denylist_examples.txt setup.cfg
+#    pip-compile --allow-unsafe --extra=pyqt5 --extra=pyqt6_experimental --extra=pyside2 --extra=pyside6_experimental --extra=testing --output-file=napari_repo/resources/constraints/constraints_py3.9_examples.txt --strip-extras napari_repo/resources/constraints/version_denylist.txt napari_repo/setup.cfg resources/constraints/version_denylist_examples.txt
 #
 alabaster==0.7.13
     # via sphinx
-app-model==0.2.0
-    # via napari (setup.cfg)
+app-model==0.2.2
+    # via napari (napari_repo/setup.cfg)
 appdirs==1.4.4
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   npe2
 asciitree==0.3.3
     # via zarr
-asttokens==2.2.1
+asttokens==2.4.0
     # via stack-data
 attrs==23.1.0
     # via
     #   hypothesis
     #   jsonschema
     #   referencing
-babel==2.12.1
+babel==2.13.0
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   sphinx
 backcall==0.2.0
     # via ipython
-build==0.10.0
+build==1.0.3
     # via npe2
 cachey==0.2.1
-    # via napari (setup.cfg)
+    # via napari (napari_repo/setup.cfg)
 certifi==2023.7.22
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   requests
-charset-normalizer==3.2.0
+charset-normalizer==3.3.0
     # via requests
-click==8.1.6
+click==8.1.7
     # via
     #   dask
     #   typer
 cloudpickle==2.2.1
     # via dask
-cmake==3.27.1
+cmake==3.27.6
     # via triton
 comm==0.1.4
     # via ipykernel
-contourpy==1.1.0
+contourpy==1.1.1
     # via matplotlib
-coverage==7.2.7
+coverage==7.3.2
     # via pytest-cov
-cycler==0.11.0
+cycler==0.12.0
     # via matplotlib
-dask==2023.8.0
-    # via napari (setup.cfg)
-debugpy==1.6.7
+dask==2023.9.3
+    # via napari (napari_repo/setup.cfg)
+debugpy==1.8.0
     # via ipykernel
 decorator==5.1.1
     # via ipython
@@ -63,116 +63,118 @@ distlib==0.3.7
     # via virtualenv
 docstring-parser==0.15
     # via magicgui
-docutils==0.17.1
+docutils==0.20.1
     # via sphinx
 entrypoints==0.4
     # via numcodecs
-exceptiongroup==1.1.2
+exceptiongroup==1.1.3
     # via
     #   hypothesis
+    #   ipython
     #   pytest
-executing==1.2.0
+executing==2.0.0
     # via stack-data
-fasteners==0.18
+fasteners==0.19
     # via zarr
-filelock==3.12.2
+filelock==3.12.4
     # via
     #   torch
     #   triton
     #   virtualenv
-fonttools==4.42.0
+fonttools==4.43.0
     # via matplotlib
 freetype-py==2.4.0
     # via vispy
-fsspec==2023.6.0
+fsspec==2023.9.2
     # via
     #   dask
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
 heapdict==1.0.1
     # via cachey
-hsluv==5.0.3
+hsluv==5.0.4
     # via vispy
-hypothesis==6.82.2
-    # via napari (setup.cfg)
+hypothesis==6.87.1
+    # via napari (napari_repo/setup.cfg)
 idna==3.4
     # via requests
-imageio==2.31.1
+imageio==2.31.5
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   napari-svg
     #   scikit-image
 imagesize==1.4.1
     # via sphinx
 importlib-metadata==6.8.0
     # via
+    #   build
     #   dask
     #   jupyter-client
     #   sphinx
-importlib-resources==6.0.1
+importlib-resources==6.1.0
     # via matplotlib
 in-n-out==0.1.8
     # via app-model
 iniconfig==2.0.0
     # via pytest
-ipykernel==6.25.0
+ipykernel==6.25.2
     # via
     #   napari-console
     #   qtconsole
-ipython==8.14.0
+ipython==8.16.1
     # via
     #   ipykernel
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   napari-console
 ipython-genutils==0.2.0
     # via qtconsole
-jedi==0.19.0
+jedi==0.19.1
     # via ipython
 jinja2==3.1.2
     # via
     #   numpydoc
     #   sphinx
     #   torch
-joblib==1.3.1
+joblib==1.3.2
     # via
     #   nilearn
     #   scikit-learn
-jsonschema==4.19.0
-    # via napari (setup.cfg)
+jsonschema==4.19.1
+    # via napari (napari_repo/setup.cfg)
 jsonschema-specifications==2023.7.1
     # via jsonschema
-jupyter-client==8.3.0
+jupyter-client==8.3.1
     # via
     #   ipykernel
     #   qtconsole
-jupyter-core==5.3.1
+jupyter-core==5.3.2
     # via
     #   ipykernel
     #   jupyter-client
     #   qtconsole
-kiwisolver==1.4.4
+kiwisolver==1.4.5
     # via
     #   matplotlib
     #   vispy
 lazy-loader==0.3
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   scikit-image
-lit==16.0.6
+lit==17.0.2
     # via triton
 locket==1.0.0
     # via partd
 lxml==4.9.3
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   nilearn
-magicgui==0.7.2
-    # via napari (setup.cfg)
+magicgui==0.7.3
+    # via napari (napari_repo/setup.cfg)
 markdown-it-py==3.0.0
     # via rich
 markupsafe==2.1.3
     # via jinja2
-matplotlib==3.7.2
-    # via napari (setup.cfg)
+matplotlib==3.8.0
+    # via napari (napari_repo/setup.cfg)
 matplotlib-inline==0.1.6
     # via
     #   ipykernel
@@ -184,17 +186,17 @@ mpmath==1.3.0
 mypy-extensions==1.0.0
     # via psygnal
 napari-console==0.0.8
-    # via napari (setup.cfg)
+    # via napari (napari_repo/setup.cfg)
 napari-plugin-engine==0.2.0
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   napari-console
     #   napari-svg
 napari-plugin-manager==0.1.0a2
-    # via napari (setup.cfg)
+    # via napari (napari_repo/setup.cfg)
 napari-svg==0.1.10
-    # via napari (setup.cfg)
-nest-asyncio==1.5.7
+    # via napari (napari_repo/setup.cfg)
+nest-asyncio==1.5.8
     # via ipykernel
 networkx==3.1
     # via
@@ -202,11 +204,11 @@ networkx==3.1
     #   torch
 nibabel==5.1.0
     # via nilearn
-nilearn==0.10.1
+nilearn==0.10.2
     # via -r resources/constraints/version_denylist_examples.txt
 npe2==0.7.2
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   napari-plugin-manager
 numcodecs==0.11.0
     # via zarr
@@ -217,13 +219,12 @@ numpy==1.23.5
     #   dask
     #   imageio
     #   matplotlib
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   napari-svg
     #   nibabel
     #   nilearn
     #   numcodecs
     #   pandas
-    #   pywavelets
     #   scikit-image
     #   scikit-learn
     #   scipy
@@ -232,8 +233,8 @@ numpy==1.23.5
     #   vispy
     #   xarray
     #   zarr
-numpydoc==1.5.0
-    # via napari (setup.cfg)
+numpydoc==1.6.0
+    # via napari (napari_repo/setup.cfg)
 nvidia-cublas-cu11==11.10.3.66
     # via
     #   nvidia-cudnn-cu11
@@ -259,7 +260,7 @@ nvidia-nccl-cu11==2.14.3
     # via torch
 nvidia-nvtx-cu11==11.7.91
     # via torch
-packaging==23.1
+packaging==23.2
     # via
     #   -r resources/constraints/version_denylist_examples.txt
     #   build
@@ -277,102 +278,102 @@ packaging==23.1
     #   superqt
     #   vispy
     #   xarray
-pandas==2.0.3 ; python_version >= "3.9"
+pandas==2.1.1 ; python_version >= "3.9"
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   nilearn
     #   xarray
 parso==0.8.3
     # via jedi
-partd==1.4.0
+partd==1.4.1
     # via dask
 pexpect==4.8.0
     # via ipython
 pickleshare==0.7.5
     # via ipython
-pillow==10.0.0
+pillow==10.0.1
     # via
     #   imageio
     #   matplotlib
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   scikit-image
 pint==0.22
-    # via napari (setup.cfg)
-platformdirs==3.10.0
+    # via napari (napari_repo/setup.cfg)
+platformdirs==3.11.0
     # via
     #   jupyter-core
     #   pooch
     #   virtualenv
-pluggy==1.2.0
+pluggy==1.3.0
     # via pytest
 pooch==1.7.0
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   scikit-image
 prompt-toolkit==3.0.39
     # via ipython
 psutil==5.9.5
     # via
     #   ipykernel
-    #   napari (setup.cfg)
-psygnal==0.9.1
+    #   napari (napari_repo/setup.cfg)
+psygnal==0.9.4
     # via
     #   app-model
     #   magicgui
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   npe2
 ptyprocess==0.7.0
     # via pexpect
 pure-eval==0.2.2
     # via stack-data
-pydantic==1.10.12
+pydantic==1.10.13
     # via
     #   app-model
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   npe2
 pygments==2.16.1
     # via
     #   ipython
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   qtconsole
     #   rich
     #   sphinx
     #   superqt
 pyopengl==3.1.6
     # via
-    #   -r resources/constraints/version_denylist.txt
-    #   napari (setup.cfg)
-pyparsing==3.0.9
+    #   -r napari_repo/resources/constraints/version_denylist.txt
+    #   napari (napari_repo/setup.cfg)
+pyparsing==3.1.1
     # via matplotlib
 pyproject-hooks==1.0.0
     # via build
 pyqt5==5.15.9
-    # via napari (setup.cfg)
+    # via napari (napari_repo/setup.cfg)
 pyqt5-qt5==5.15.2
     # via pyqt5
 pyqt5-sip==12.12.2
     # via pyqt5
 pyqt6==6.5.2
-    # via napari (setup.cfg)
+    # via napari (napari_repo/setup.cfg)
 pyqt6-qt6==6.5.2
     # via pyqt6
 pyqt6-sip==13.5.2
     # via pyqt6
 pyside2==5.15.2.1 ; python_version != "3.8"
-    # via napari (setup.cfg)
+    # via napari (napari_repo/setup.cfg)
 pyside6==6.3.1 ; python_version < "3.10"
     # via
-    #   -r resources/constraints/version_denylist.txt
-    #   napari (setup.cfg)
+    #   -r napari_repo/resources/constraints/version_denylist.txt
+    #   napari (napari_repo/setup.cfg)
 pyside6-addons==6.3.1
     # via pyside6
 pyside6-essentials==6.3.1
     # via
     #   pyside6
     #   pyside6-addons
-pytest==7.4.0
+pytest==7.4.2
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   pytest-cov
     #   pytest-json-report
     #   pytest-metadata
@@ -380,43 +381,41 @@ pytest==7.4.0
     #   pytest-qt
 pytest-cov==4.1.0
     # via
-    #   -r resources/constraints/version_denylist.txt
-    #   napari (setup.cfg)
+    #   -r napari_repo/resources/constraints/version_denylist.txt
+    #   napari (napari_repo/setup.cfg)
 pytest-json-report==1.5.0
-    # via -r resources/constraints/version_denylist.txt
+    # via -r napari_repo/resources/constraints/version_denylist.txt
 pytest-metadata==3.0.0
     # via pytest-json-report
 pytest-pretty==1.2.0
-    # via napari (setup.cfg)
+    # via napari (napari_repo/setup.cfg)
 pytest-qt==4.2.0
-    # via napari (setup.cfg)
+    # via napari (napari_repo/setup.cfg)
 python-dateutil==2.8.2
     # via
     #   jupyter-client
     #   matplotlib
     #   pandas
-pytz==2023.3
+pytz==2023.3.post1
     # via pandas
-pywavelets==1.4.1
-    # via scikit-image
 pyyaml==6.0.1
     # via
     #   dask
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   npe2
-pyzmq==25.1.0
+pyzmq==25.1.1
     # via
     #   ipykernel
     #   jupyter-client
     #   qtconsole
-qtconsole==5.4.3
+qtconsole==5.4.4
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   napari-console
-qtpy==2.3.1
+qtpy==2.4.0
     # via
     #   magicgui
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   napari-console
     #   napari-plugin-manager
     #   qtconsole
@@ -430,22 +429,22 @@ requests==2.31.0
     #   nilearn
     #   pooch
     #   sphinx
-rich==13.5.2
+rich==13.6.0
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   npe2
     #   pytest-pretty
-rpds-py==0.9.2
+rpds-py==0.10.3
     # via
     #   jsonschema
     #   referencing
-scikit-image==0.21.0
-    # via napari (setup.cfg)
-scikit-learn==1.3.0
+scikit-image==0.22.0
+    # via napari (napari_repo/setup.cfg)
+scikit-learn==1.3.1
     # via nilearn
-scipy==1.11.1 ; python_version >= "3.9"
+scipy==1.11.3 ; python_version >= "3.9"
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   nilearn
     #   scikit-image
     #   scikit-learn
@@ -464,46 +463,53 @@ snowballstemmer==2.2.0
     # via sphinx
 sortedcontainers==2.4.0
     # via hypothesis
-sphinx==4.5.0
+sphinx==7.2.6
     # via
-    #   napari (setup.cfg)
     #   numpydoc
-sphinxcontrib-applehelp==1.0.4
+    #   sphinxcontrib-applehelp
+    #   sphinxcontrib-devhelp
+    #   sphinxcontrib-htmlhelp
+    #   sphinxcontrib-qthelp
+    #   sphinxcontrib-serializinghtml
+sphinxcontrib-applehelp==1.0.7
     # via sphinx
-sphinxcontrib-devhelp==1.0.2
+sphinxcontrib-devhelp==1.0.5
     # via sphinx
-sphinxcontrib-htmlhelp==2.0.1
+sphinxcontrib-htmlhelp==2.0.4
     # via sphinx
 sphinxcontrib-jsmath==1.0.1
     # via sphinx
-sphinxcontrib-qthelp==1.0.3
+sphinxcontrib-qthelp==1.0.6
     # via sphinx
-sphinxcontrib-serializinghtml==1.1.5
+sphinxcontrib-serializinghtml==1.1.9
     # via sphinx
-stack-data==0.6.2
+stack-data==0.6.3
     # via ipython
-superqt==0.5.0
+superqt==0.6.0
     # via
     #   magicgui
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   napari-plugin-manager
 sympy==1.12
     # via torch
-tensorstore==0.1.40
+tabulate==0.9.0
+    # via numpydoc
+tensorstore==0.1.45
     # via
-    #   -r resources/constraints/version_denylist.txt
-    #   napari (setup.cfg)
+    #   -r napari_repo/resources/constraints/version_denylist.txt
+    #   napari (napari_repo/setup.cfg)
 threadpoolctl==3.2.0
     # via scikit-learn
-tifffile==2023.7.18
+tifffile==2023.9.26
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   scikit-image
 tomli==2.0.1
     # via
     #   build
     #   coverage
     #   npe2
+    #   numpydoc
     #   pyproject-hooks
     #   pytest
 tomli-w==1.0.0
@@ -511,19 +517,19 @@ tomli-w==1.0.0
 toolz==0.12.0
     # via
     #   dask
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   partd
 torch==2.0.1
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   triton
-tornado==6.3.2
+tornado==6.3.3
     # via
     #   ipykernel
     #   jupyter-client
-tqdm==4.65.0
-    # via napari (setup.cfg)
-traitlets==5.9.0
+tqdm==4.66.1
+    # via napari (napari_repo/setup.cfg)
+traitlets==5.11.2
     # via
     #   comm
     #   ipykernel
@@ -536,12 +542,12 @@ triton==2.0.0
     # via torch
 typer==0.9.0
     # via npe2
-typing-extensions==4.7.1
+typing-extensions==4.8.0
     # via
     #   app-model
     #   ipython
     #   magicgui
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   pint
     #   psygnal
     #   pydantic
@@ -550,17 +556,17 @@ typing-extensions==4.7.1
     #   typer
 tzdata==2023.3
     # via pandas
-urllib3==2.0.4
+urllib3==2.0.6
     # via requests
-virtualenv==20.24.2
-    # via napari (setup.cfg)
-vispy==0.13.0
+virtualenv==20.24.5
+    # via napari (napari_repo/setup.cfg)
+vispy==0.14.1
     # via
-    #   napari (setup.cfg)
+    #   napari (napari_repo/setup.cfg)
     #   napari-svg
-wcwidth==0.2.6
+wcwidth==0.2.8
     # via prompt-toolkit
-wheel==0.41.1
+wheel==0.41.2
     # via
     #   nvidia-cublas-cu11
     #   nvidia-cuda-cupti-cu11
@@ -569,12 +575,12 @@ wheel==0.41.1
     #   nvidia-cusparse-cu11
     #   nvidia-nvtx-cu11
 wrapt==1.15.0
-    # via napari (setup.cfg)
-xarray==2023.7.0
-    # via napari (setup.cfg)
-zarr==2.16.0
-    # via napari (setup.cfg)
-zipp==3.16.2
+    # via napari (napari_repo/setup.cfg)
+xarray==2023.9.0
+    # via napari (napari_repo/setup.cfg)
+zarr==2.16.1
+    # via napari (napari_repo/setup.cfg)
+zipp==3.17.0
     # via
     #   importlib-metadata
     #   importlib-resources
@@ -582,7 +588,7 @@ zipp==3.16.2
 # The following packages are considered to be unsafe in a requirements file:
 pip==23.2.1
     # via napari-plugin-manager
-setuptools==68.0.0
+setuptools==68.2.2
     # via
     #   nvidia-cublas-cu11
     #   nvidia-cuda-cupti-cu11
diff --git a/setup.cfg b/setup.cfg
index f37ea199..39dffc1f 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -78,7 +78,7 @@ install_requires =
     toolz>=0.10.0
     tqdm>=4.56.0
     typing_extensions>=4.2.0
-    vispy>=0.13.0,<0.14
+    vispy>=0.14.1,<0.15
     wrapt>=1.11.1


--
2.34.1
